passes/sat/sim.cc

```diff
@@ -2397,6 +2397,170 @@ struct VCDWriter : public OutputWriter
 	std::ofstream vcdfile;
 };
 
+struct AnnotateActivity : public OutputWriter {
+	AnnotateActivity(SimWorker *worker) : OutputWriter(worker) {}
+
+	struct SignalActivityData {
+		std::vector<uint32_t> lastValues;
+		std::vector<uint32_t> toggleCounts;
+		std::vector<uint32_t> highTimes;
+	};
+
+	typedef std::map<int, SignalActivityData> SignalActivityDataMap;
+
+	void write(std::map<int, bool> &use_signal) override
+	{
+		// Init map
+		SignalActivityDataMap dataMap;
+		// For each event (new time when a value changed)
+		for (auto &d : worker->output_data) {
+			// For each signal/values in that time slice
+			for (auto &data : d.second) {
+				int sig = data.first;
+				if (!use_signal.at(sig))
```

Comment:
Can you add a comment for this? Don't understand it

Reply:
Which line(s)? Hard to tell from where the comment is inserted.

Reply:
Sorry. It's always the line right above the comment. So:
```
if (!use_signal.at(sig))
```

Not sure what this is?

Reply:
Oh, it's coming from the sim pass, they have that for all the writers, just following the pattern

Reply:
Do you know what it is? Maybe it's to allow selection of a subset of signals for VCD reading/dumping?

Reply:
It is created here:
https://github.com/Silimate/yosys/blob/650fc2801d781894d3abcb0f5c0a9f229b2de13f/passes/sat/sim.cc#L1322
It's used for all the post sim writers. Some filtering of the VCD data.